### ISSUE TITLE
Fix reforging

### ISSUE THREAD

**Issue by Itorius** [tModLoader/tModLoader#1101]
```
Chicken-Bones:
While you're at it, can we fix the mess which is reforging?
I suggest adding a method on Item called ResetStats which generates a new Item with SetDefaults and then copies over the damage, useAnimation, useTime, reuseDelay, mana, knockBack, scale, shootSpeed, crit, value fields. Then add the same hook in Mod/GlobalItem, and pass the 'default item' for reference
```

**Comment by Chicken-Bones** [comment]
Needs fix with #1131

**Comment by Yoraiz0r** [comment]
As per discussions, it looks like the primary issue with reforging right now is - it is impossible know which data should be transferred due to a lack of a proper hook.
A hook like `TransferDataTo(Item other, IItemDataTransferReason reason)` , with a default implementation of cloning the mod-data, and applied to both the ModItem and GlobalItems of the item, sounds ideal, as it would allow partial (or no) transfers without breaking the current functionality.

**Comment by Chicken-Bones** [comment]
This was fixed in the 1.4.4 port
